File: easy/118_Pascal's_Triangle/main.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans(numRows);
        
        if(numRows > 0){
            ans[0] = {1};
        }

        if(numRows > 1){
            ans[1] = {1,1};
        }

        for(int i = 2; i < numRows; i++){
            vector<int> currRow(i+1, 1);
    
            for(int j = 1; j < i; j++){
                currRow[j] = ans[i-1][j-1] + ans[i-1][j];
            }

            ans[i] = currRow;
        }
       
        return ans;
    }
};

File: easy/118_Pascal's_Triangle/main.py
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []

        if numRows > 0:
            ans.append([1])

        for i in range(1, numRows):
            prev = ans[i-1]
            cur = [1]

            for j in range(1, i):
                cur.append(prev[j-1] + prev[j])

            cur.append(1)
            ans.append(cur)

        return ans